### ISSUE TITLE
Native module WebRTCModule tried to override WebRTCModule

### ISSUE THREAD

**Issue by Lk2v** [react-native-webrtc/react-native-webrtc#785]
I followed the Android installation instructions but I got this error I checked and the package is declared only once.

![image](https://user-images.githubusercontent.com/26687216/81808106-82405380-951f-11ea-857b-ef1daf53345d.png)

#### MainApplication.java
```Java
package ...;

import android.app.Application;
import android.content.Context;
import com.facebook.react.PackageList;
import com.facebook.react.ReactApplication;
import com.learnium.RNDeviceInfo.RNDeviceInfo;
import com.swmansion.reanimated.ReanimatedPackage;
import org.reactnative.camera.RNCameraPackage;
import com.oney.WebRTCModule.WebRTCModulePackage;
import com.facebook.react.ReactInstanceManager;
import com.facebook.react.ReactNativeHost;
import com.facebook.react.ReactPackage;
import com.facebook.soloader.SoLoader;
import java.lang.reflect.InvocationTargetException;
import java.util.List;

public class MainApplication extends Application implements ReactApplication {

  private final ReactNativeHost mReactNativeHost =
      new ReactNativeHost(this) {
        @Override
        public boolean getUseDeveloperSupport() {
          return BuildConfig.DEBUG;
        }

        @Override
        protected List<ReactPackage> getPackages() {
          @SuppressWarnings("UnnecessaryLocalVariable")
          List<ReactPackage> packages = new PackageList(this).getPackages();
          // Packages that cannot be autolinked yet can be added manually here, for example:
          // packages.add(new MyReactNativePackage());
          packages.add(new WebRTCModulePackage());
          
          return packages;
        }

        @Override
        protected String getJSMainModuleName() {
          return "index";
        }

        
      };

  @Override
  public ReactNativeHost getReactNativeHost() {
    return mReactNativeHost;
  }

  @Override
  public void onCreate() {
    super.onCreate();
    SoLoader.init(this, /* native exopackage */ false);
    initializeFlipper(this, getReactNativeHost().getReactInstanceManager());
  }

  /**
   * Loads Flipper in React Native templates. Call this in the onCreate method with something like
   * initializeFlipper(this, getReactNativeHost().getReactInstanceManager());
   *
   * @param context
   * @param reactInstanceManager
   */
  private static void initializeFlipper(
      Context context, ReactInstanceManager reactInstanceManager) {
    if (BuildConfig.DEBUG) {
      try {
        /*
         We use reflection here to pick up the class that initializes Flipper,
        since Flipper library is not available in release mode
        */
        Class<?> aClass = Class.forName("com.pipelinetransfert.ReactNativeFlipper");
        aClass
            .getMethod("initializeFlipper", Context.class, ReactInstanceManager.class)
            .invoke(null, context, reactInstanceManager);
      } catch (ClassNotFoundException e) {
        e.printStackTrace();
      } catch (NoSuchMethodException e) {
        e.printStackTrace();
      } catch (IllegalAccessException e) {
        e.printStackTrace();
      } catch (InvocationTargetException e) {
        e.printStackTrace();
      }
    }
  }
}
```
#### Platform information

* **React Native version**: 0.62.2
* **react-native-webrtc version**: 1.75.3
* **OS**: Android
* **OS version**: 9

**Comment by 8BallBomBom** [comment]
As you are running React-Native 62.2 you don't need to follow the manual instructions and you also don't need to use auto linking either.
Pretty much once you add the package via yarn or npm, the linking for Android and iOS gets done automatically.

**Comment by Lk2v** [comment]
Thank you for your quick response !
I started a new react-native project to test this but the application crashes at startup same problem than : #780 

![image](https://user-images.githubusercontent.com/26687216/81820271-ae180500-9530-11ea-944f-7523f135b6e6.png)

**Comment by RomanSuvorov** [comment]
@Lk2v 

did you solve the problem of launching the application without application crashes at startup?

**Comment by Lk2v** [comment]
> @Lk2v
> 
> Did you solve the problem of launching the application without application crashes at startup?

No, unfortunately, I haven't solved the problem.
However with the installation instructions for Android (+ by fixing some bugs) I succeeded to launch the application without crashing but I got the error (image of the first post) maybe it could work if this error could be fixed.

**Comment by 8BallBomBom** [comment]
You shouldn't get that error if you only add the module via npm or yarn without following the extra install steps as they are no longer needed on newer React-Native versions.
I can't reproduce the errors and crashes on my side.

Have you tried getting logs from logcat?

**Comment by Lk2v** [comment]
Here are the logs I got while listening with `react-native log-android` :
```Shell
info Starting logkitty
[09:48:19] D | ReactNative ▶︎ ReactInstanceManager.ctor()

[09:48:19] D | ReactNative ▶︎ ReactInstanceManager.createReactContextInBackground()
                           └ ReactInstanceManager.recreateReactContextInBackgroundInner()

[09:48:21] D | ReactNative ▶︎ ReactInstanceManager.onJSBundleLoadedFromServer()

[09:48:21] D | ReactNative ▶︎ ReactInstanceManager.recreateReactContextInBackground()
                           └ ReactInstanceManager.runCreateReactContextOnNewThread()

[09:48:21] D | ReactNative ▶︎ ReactInstanceManager.createReactContext()

[09:48:21] D | ReactNative ▶︎ Initializing React Xplat Bridge.

[09:48:21] D | ReactNative ▶︎ Initializing React Xplat Bridge before initializeBridge

[09:48:21] D | ReactNative ▶︎ Initializing React Xplat Bridge after initializeBridge
                           └ CatalystInstanceImpl.runJSBundle()
--------- beginning of crash

[09:48:21] D | ReactNative ▶︎ ReactInstanceManager.setupReactContext()

[09:48:21] D | ReactNative ▶︎ CatalystInstanceImpl.initialize()

[09:48:21] D | ReactNative ▶︎ ReactInstanceManager.attachRootViewToInstance()

[09:48:21] D | ReactNative ▶︎ ReactInstanceManager.detachViewFromInstance()

[09:48:22] I | ReactNativeJS ▶︎ Running "RTCTest" with {"rootTag":1}
```

**Comment by Lk2v** [comment]
And here is the error I get when I listen using adb logcat :

```Shell
com.rtctest: Not late-enabling -Xcheck:jni (already on)
05-14 10:07:11.470  1703  1969 D         : HostConnection::get() New Host Connection established 0xf03167c0, tid 1969
com.rtctest: Accessing hidden method Lcom/android/org/conscrypt/OpenSSLSocketImpl;->setUseSessionTickets(Z)V (light greylist, reflection)
05-14 10:07:11.812 23860 23860 W com.rtctest: Accessing hidden method Lcom/android/org/conscrypt/OpenSSLSocketImpl;->setHostname(Ljava/lang/String;)V (light greylist, reflection)    
05-14 10:07:11.812 23860 23860 W com.rtctest: Accessing hidden method Lcom/android/org/conscrypt/OpenSSLSocketImpl;->getAlpnSelectedProtocol()[B (light greylist, reflection)
05-14 10:07:11.812 23860 23860 W com.rtctest: Accessing hidden method Lcom/android/org/conscrypt/OpenSSLSocketImpl;->setAlpnProtocols([B)V (light greylist, reflection)
05-14 10:07:11.812 23860 23860 W com.rtctest: Accessing hidden method Ldalvik/system/CloseGuard;->get()Ldalvik/system/CloseGuard; (light greylist, reflection)
05-14 10:07:11.812 23860 23860 W com.rtctest: Accessing hidden method Ldalvik/system/CloseGuard;->open(Ljava/lang/String;)V (light greylist, reflection)
05-14 10:07:11.812 23860 23860 W com.rtctest: Accessing hidden method Ldalvik/system/CloseGuard;->warnIfOpen()V (light greylist, reflection)

com.rtctest: Accessing hidden field Landroid/view/View;->mKeyedTags:Landroid/util/SparseArray; (light greylist, reflection)
05-14 10:07:11.927 23860 23860 W com.rtctest: Accessing hidden field Landroid/view/View;->mListenerInfo:Landroid/view/View$ListenerInfo; (light greylist, reflection)
05-14 10:07:11.927 23860 23860 W com.rtctest: Accessing hidden field Landroid/view/View$ListenerInfo;->mOnClickListener:Landroid/view/View$OnClickListener; (light greylist, reflection)


com.rtctest: Accessing hidden method Landroid/graphics/drawable/Drawable;->getOpticalInsets()Landroid/graphics/Insets; (light greylist, linking)
05-14 10:07:12.039 23860 23860 W com.rtctest: Accessing hidden field Landroid/graphics/Insets;->left:I (light greylist, linking)
05-14 10:07:12.039 23860 23860 W com.rtctest: Accessing hidden field Landroid/graphics/Insets;->right:I (light greylist, linking)
05-14 10:07:12.039 23860 23860 W com.rtctest: Accessing hidden field Landroid/graphics/Insets;->top:I (light greylist, linking)
05-14 10:07:12.039 23860 23860 W com.rtctest: Accessing hidden field Landroid/graphics/Insets;->bottom:I (light greylist, linking)

com.rtctest: Accessing hidden method Landroid/view/View;->computeFitSystemWindows(Landroid/graphics/Rect;Landroid/graphics/Rect;)Z (light greylist, reflection)
05-14 10:07:12.103 23860 23860 W com.rtctest: Accessing hidden method Landroid/view/ViewGroup;->makeOptionalFitsSystemWindows()V (light greylist, reflection)

05-14 10:07:12.292 23860 23860 W com.rtctest: Accessing hidden method Landroid/widget/TextView;->getTextDirectionHeuristic()Landroid/text/TextDirectionHeuristic; (light greylist, linking)

05-14 10:07:13.292 23860 23899 D ReactNative: Initializing React Xplat Bridge.
05-14 10:07:13.292 23860 23900 I org.webrtc.Logging: NativeLibrary: Loading native library: jingle_peerconnection_so
05-14 10:07:13.292 23860 23900 I org.webrtc.Logging: NativeLibrary: Loading library: jingle_peerconnection_so
05-14 10:07:13.307 23860 23899 D ReactNative: Initializing React Xplat Bridge before initializeBridge
05-14 10:07:13.326 23860 23900 I org.webrtc.Logging: PeerConnectionFactory: PeerConnectionFactory was initialized without an injected Loggable. Any existing Loggable will be deleted.05-14 10:07:13.329 23860 23899 D ReactNative: Initializing React Xplat Bridge after initializeBridge
05-14 10:07:13.329 23860 23899 D ReactNative: CatalystInstanceImpl.runJSBundle()
05-14 10:07:13.329 23860 23900 E AndroidRuntime: FATAL EXCEPTION: pool-1-thread-1
05-14 10:07:13.329 23860 23900 E AndroidRuntime: Process: com.rtctest, PID: 23860
05-14 10:07:13.329 23860 23900 E AndroidRuntime: java.lang.NoSuchMethodError: No static method createEgl14([I)Lorg/webrtc/EglBase14; in class Lorg/webrtc/EglBase; or its super classes (declaration of 'org.webrtc.EglBase' appears in /data/app/com.rtctest--mbvvPVeW0Fi05tovQcx_g==/base.apk)
05-14 10:07:13.329 23860 23900 E AndroidRuntime:        at com.oney.WebRTCModule.EglUtils.getRootEglBase(EglUtils.java:35)
05-14 10:07:13.329 23860 23900 E AndroidRuntime:        at com.oney.WebRTCModule.EglUtils.getRootEglBaseContext(EglUtils.java:62)
05-14 10:07:13.329 23860 23900 E AndroidRuntime:        at com.oney.WebRTCModule.WebRTCModule.initAsync(WebRTCModule.java:98)
05-14 10:07:13.329 23860 23900 E AndroidRuntime:        at com.oney.WebRTCModule.WebRTCModule.lambda$new$0$WebRTCModule(WebRTCModule.java:73)
05-14 10:07:13.329 23860 23900 E AndroidRuntime:        at com.oney.WebRTCModule.-$$Lambda$WebRTCModule$CnyHZvkjDxq52UReGHUZlY0JsVw.run(Unknown Source:4)
05-14 10:07:13.329 23860 23900 E AndroidRuntime:        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
05-14 10:07:13.329 23860 23900 E AndroidRuntime:        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
05-14 10:07:13.329 23860 23900 E AndroidRuntime:        at java.lang.Thread.run(Thread.java:764)
05-14 10:07:13.330 23860 23902 D ReactNative: ReactInstanceManager.setupReactContext()
05-14 10:07:13.331 23860 23902 D ReactNative: CatalystInstanceImpl.initialize()
05-14 10:07:13.333  1860  3787 W ActivityManager:   Force finishing activity com.rtctest/.MainActivity
05-14 10:07:13.345  1860  1876 I ActivityManager: Showing crash dialog for package com.rtctest u0
05-14 10:07:13.347 23860 23902 D ReactNative: ReactInstanceManager.attachRootViewToInstance()
05-14 10:07:13.388  1860  1875 W BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.stats.service.DropBoxEntryAddedReceiver
05-14 10:07:13.388  1860  1875 W BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.chimera.GmsIntentOperationService$PersistentTrustedReceiver
05-14 10:07:13.403 10375 22547 I PBSessionCacheImpl: Deleted sessionId[977967182212] from persistence.
05-14 10:07:13.407  1605  1605 D gralloc_ranchu: gralloc_alloc: Creating ashmem region of size 3432448
05-14 10:07:13.416 10375 10375 W SearchService: Abort, client detached.


05-14 10:07:13.436 23860 23902 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.art.ARTShapeViewManager
05-14 10:07:13.436 23860 23902 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.art.ARTShapeShadowNode
05-14 10:07:13.437  1860  4162 D EGL_emulation: eglCreateContext: 0xd1524ea0: maj 3 min 0 rcv 3
05-14 10:07:13.439 23860 23902 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.art.ARTTextViewManager
05-14 10:07:13.440 23860 23902 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.art.ARTTextShadowNode
05-14 10:07:13.441  1605  1605 D gralloc_ranchu: gralloc_alloc: Creating ashmem region of size 9334784
05-14 10:07:13.441 23860 23902 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.checkbox.ReactCheckBoxManager
05-14 10:07:13.441  1860  4162 D EGL_emulation: eglMakeCurrent: 0xd1524ea0: ver 3 0 (tinfo 0xc6aebd20)
05-14 10:07:13.450  1605  1666 D gralloc_ranchu: gralloc_alloc: Creating ashmem region of size 9334784
05-14 10:07:13.455 23860 23902 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.uimanager.LayoutShadowNode
05-14 10:07:13.461  1605  1666 D gralloc_ranchu: gralloc_alloc: Creating ashmem region of size 9334784
05-14 10:07:13.466 23860 23902 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.picker.ReactDialogPickerManager
05-14 10:07:13.470 23860 23902 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.drawer.ReactDrawerLayoutManager
05-14 10:07:13.472  1605  1666 D gralloc_ranchu: gralloc_alloc: Creating ashmem region of size 9334784
05-14 10:07:13.475 23860 23902 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.picker.ReactDropdownPickerManager
05-14 10:07:13.479 23860 23902 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.scroll.ReactHorizontalScrollViewManager       
05-14 10:07:13.480  1605  1666 D gralloc_ranchu: gralloc_alloc: Creating ashmem region of size 9334784
05-14 10:07:13.490 23860 23902 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.scroll.ReactHorizontalScrollContainerViewManager
05-14 10:07:13.492 23860 23902 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.progressbar.ReactProgressBarViewManager       
05-14 10:07:13.495  1605  1666 D gralloc_ranchu: gralloc_alloc: Creating ashmem region of size 9334784
05-14 10:07:13.495 23860 23902 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.progressbar.ProgressBarShadowNode
05-14 10:07:13.499 23860 23902 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.scroll.ReactScrollViewManager
05-14 10:07:13.508 23860 23902 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.slider.ReactSliderManager
05-14 10:07:13.510 23860 23902 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.slider.ReactSliderManager$ReactSliderShadowNode
05-14 10:07:13.512 23860 23902 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.switchview.ReactSwitchManager
05-14 10:07:13.515 23860 23902 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.switchview.ReactSwitchManager$ReactSwitchShadowNode
05-14 10:07:13.515  2517  2640 D EGL_emulation: eglMakeCurrent: 0xd62217a0: ver 3 0 (tinfo 0xd6218880)
05-14 10:07:13.516 23860 23902 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.swiperefresh.SwipeRefreshLayoutManager        
05-14 10:07:13.519 23860 23902 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.art.ARTSurfaceViewManager
05-14 10:07:13.522 10375 10471 D EGL_emulation: eglMakeCurrent: 0xf17c17c0: ver 3 0 (tinfo 0xe8dedac0)
05-14 10:07:13.523 23860 23902 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.art.ARTSurfaceViewShadowNode
05-14 10:07:13.524 23860 23902 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.text.frescosupport.FrescoBasedReactTextInlineImageViewManager
05-14 10:07:13.525 23860 23902 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.text.frescosupport.FrescoBasedReactTextInlineImageShadowNode
05-14 10:07:13.529 23860 23902 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.image.ReactImageManager
05-14 10:07:13.533 23860 23902 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.modal.ReactModalHostManager
05-14 10:07:13.534 23860 23902 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.modal.ModalHostShadowNode
05-14 10:07:13.535 23860 23902 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.text.ReactRawTextManager
05-14 10:07:13.535 23860 23902 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.text.ReactRawTextShadowNode
05-14 10:07:13.544 23860 23902 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.textinput.ReactTextInputManager
05-14 10:07:13.555 23860 23902 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.textinput.ReactTextInputShadowNode
05-14 10:07:13.571 23860 23902 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.text.ReactTextViewManager
```

**Comment by 8BallBomBom** [comment]
It would seek that crash error is related to [this](https://github.com/react-native-webrtc/react-native-webrtc/issues/720).

**Comment by Lk2v** [comment]
It worked !
For those who have the same problem I set `minSdkVersion = 24` in _build.gradle_ and I put `android.enableDexingArtifactTransform.desugaring=false` in `app/gradle.properties`
I think this could fix the problem for #780 

Thank you very much @8BallBomBom !

**Comment by 8BallBomBom** [comment]
You should still be able to maintain a minsdk of 16 as long as there aren't any other modules forcing it to be higher.

Glad things are working :)
